### PR TITLE
Taking into account the sticky scroll widget when revealing a range

### LINKED ISSUE #157175: Revealing in editor is not aware of sticky lines
It should be either disabled or maybe better have very limited context (a single line max?)

Find references:

Screen Shot 2022-08-04 at 5 01 05 pm

### PR DESCRIPTION
**Pull request by aiday-mar**
Fixes #157175

The new solution does not use the sticky widget height to reveal ranges. Three cases are tested and shown in the gif:

1. Moving the cursor up by one using the arrow pointing up on the keyboard
2. Undoing and revealing the line where the last operation was performed
3. Continuing to write on a line outside of the visible view port and revealing that line 

sticky-scroll-ts

### CODE REVIEW

File: src/vs/editor/common/config/editorOptions.ts [hunk 1]
```diff
@@ -2523,19 +2523,21 @@ export interface IEditorExperimentalOptions {
 		 * Enable the sticky scroll
 		 */
 		enabled?: boolean;
+		numberLines?: number;
```
Comment (alexdima) [inline src/vs/editor/common/config/editorOptions.ts:2526]: How about we rename this to something like `maxLineCount`? `numberLines` could be interpreted as always having a certain number of lines.

File: src/vs/editor/browser/viewParts/lines/viewLines.ts [hunk 2]
```diff
@@ -718,6 +731,10 @@ export class ViewLines extends ViewPart implements IVisibleLinesHost<ViewLine>,
 			newScrollTop = this._computeMinimumScrolling(viewportStartY, viewportEndY, boxStartY, boxEndY, verticalType === viewEvents.VerticalRevealType.Top, verticalType === viewEvents.VerticalRevealType.Bottom);
 		}
 
+		const scrollTopLineNumber = Math.ceil(newScrollTop / this._lineHeight);
+		if (this._stickyScrollEnabled && range && range?.startLineNumber - scrollTopLineNumber <= this._maxNumberStickyLines) {
+			return newScrollTop + this._lineHeight * (range?.startLineNumber - scrollTopLineNumber - this._maxNumberStickyLines - 1);
+		}
```
Comment (alexdima) [inline src/vs/editor/browser/viewParts/lines/viewLines.ts:737]: Could we remove this logic from here and adjust `boxStartY` above in the same area where `cursorSurroundingLines` is handled? The code above needs to be tweaked, maybe we introduce two variables, `paddingTop` and `paddingBottom` and lines 681-697 can be modified to define `paddingTop`/`paddingBottom` and then those can be subtracted/added to `boxStartY`/`boxEndY`.

File: src/vs/editor/common/config/editorOptions.ts [hunk 3]
```diff
@@ -2544,6 +2546,17 @@ class EditorExperimental extends BaseEditorOption<EditorOption.experimental, IEd
 					default: defaults.stickyScroll.enabled,
 					description: nls.localize('editor.experimental.stickyScroll', "Shows the nested current scopes during the scroll at the top of the editor.")
 				},
+				'editor.experimental.stickyScroll.numberLines': {
+					type: 'number',
+					default: defaults.stickyScroll.numberLines,
+					enum: [1, 5, 10],
```
Comment (alexdima) [inline src/vs/editor/common/config/editorOptions.ts:2552]: This doesn't need to be an enum, it can be a regular number.

**Review by alexdima — CHANGES_REQUESTED** [review]
I left a few comments

**Review by alexdima — APPROVED** [review]

**Comment by keybraker** [comment]
Does this also solve the search problem and top right filtering options that get hidden?

**Pull request merged**